src/pages/reset-password/reset-password.component.js

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import { useParams } from 'react-router-dom';
+import CssBaseline from '@material-ui/core/CssBaseline';
+import LockOutlinedIcon from '@material-ui/icons/LockOutlined';
+import Typography from '@material-ui/core/Typography';
+import Container from '@material-ui/core/Container';
+import CircularProgress from '@material-ui/core/CircularProgress';
+import * as yup from 'yup';
+import * as S from './reset-password.styles';
+
+import { useMutationNewPassword } from 'service/password';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { useForm } from 'react-hook-form';
+import RHFInput from 'components/rhf-input/rhf-input.component';
+
+const schema = yup.object().shape({
+  password: yup.string().required('Campo obrigatório'),
+  confirmPassword: yup
+    .string()
+    .oneOf(
+      [yup.ref('password'), null],
+      'As senhas não são iguais. Tente novamente.'
+    )
+    .required('Campo obrigatório'),
+});
+
+const ResetPassword = () => {
+  const { handleSubmit, control } = useForm({
```

Comment:
You can have a custom hook to abstract all of this including those functions you placed in the service file.